### ISSUE TITLE
dockerPrepare should not depend on dockerClean

### ISSUE THREAD

**Issue by deanorderly** [palantir/gradle-docker#354]
gradle is meant to detect changes and only run the task if there is changes in source.  doing a dockerClean sort of defeats the purpose of that.

**Comment by tkgregory** [comment]
Agreed. Currently the `dockerPrepare` task is executing every time since it's detecting that it's outputs have been removed. Unless there's something I'm missing, I suggest hooking `dockerClean` instead as a `dependsOn` for the `clean` task, if it exists.
